openpgp/packet/signature.go

```diff
@@ -160,7 +160,10 @@ func (sig *Signature) parse(r io.Reader) (err error) {
 	switch sig.PubKeyAlgo {
 	case PubKeyAlgoRSA, PubKeyAlgoRSASignOnly:
 		sig.RSASignature = new(encoding.MPI)
-		_, err = sig.RSASignature.ReadFrom(r)
+		if _, err = sig.RSASignature.ReadFrom(r); err != nil {
+			return
+		}
+
```

Comment:
I'm confused however, the previous code was already writing to `err`, which is a return parameter, and directly below this `switch` is a `return` - so this shouldn't change anything, no?

Reply:
It just seemed neater to me, only because other cases had returns (because they have 2 signature values) but yes, it's the same thing, Reverted this one in next commit.